src/formatter-settings/index.ts

```diff
@@ -4,27 +4,347 @@
+                    id: id,
+                    value: FormatterConverter.profile2WebViewConvert(id, value),
+                });
+            }
+            this.formatWithProfileSettings(webviewPanel);
+        });
+
+        webviewPanel.onDidDispose(() => {
+            changeDocumentSubscription.dispose();
+        });
+
+        webviewPanel.webview.onDidReceiveMessage(async (e) => {
+            switch (e.command) {
+                case "onWillChangeExampleKind": {
+                    this.exampleKind = e.exampleKind;
+                    this.formatWithProfileSettings(webviewPanel);
+                    break;
+                }
+                case "onWillChangeSetting": {
+                    const settingValue: string | undefined = FormatterConverter.webView2ProfileConvert(e.id, e.value.toString());
+                    if (!settingValue) {
+                        return;
+                    }
+                    if (SupportedSettings.commonSettings.includes(e.id)) {
+                        const config = vscode.workspace.getConfiguration(undefined, { languageId: "java" });
+                        if (e.id === SupportedSettings.TABULATION_CHAR) {
+                            const targetValue = (settingValue === "tab") ? false : true;
+                            await config.update("editor.insertSpaces", targetValue, undefined, true);
+                        } else if (e.id === SupportedSettings.TABULATION_SIZE) {
+                            await config.update("editor.tabSize", Number(settingValue), undefined, true);
+                        }
+                        this.profileSettings.set(e.id, settingValue);
+                        webviewPanel.webview.postMessage({
+                            command: "changeSetting",
+                            id: e.id,
+                            value: e.value,
+                        });
+                        this.formatWithProfileSettings(webviewPanel);
+                    } else {
+                        await this.changeProfileSetting(e.id, settingValue, document);
+                    }
+                    break;
+                }
+                default:
+                    break;
+            }
+        });
+    }
+
+    private async initializeFromProfile(document: vscode.TextDocument, webviewPanel: vscode.WebviewPanel) {
+        this.profileElements.clear();
+        this.profileSettings.clear();
+        this.profileSettingsArray = [];
+        const documentDOM = new DOMParser().parseFromString(document.getText());
+        const profilesVersion = documentDOM.documentElement.getAttribute("version");
+        if (profilesVersion) {
+            this.settingsVersion = profilesVersion;
+        }
+        const profiles = documentDOM.documentElement.getElementsByTagName("profile");
+        if (!profiles || profiles.length === 0) {
+            throw new Error("No available profiles exist");
+        }
+        for (let i = 0; i < profiles.length; i++) {
```

Comment:
Why we use index-loop here?

Reply:
Since profiles are [HTMLCollection](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCollection), which doesn't support for-of loop, we have to change it to array by `Array.(profiles)` to support for-of loop, I'm afraid of the translating time. So I use index-loop here.